model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -184,55 +187,72 @@ message ProcessBundleDescriptor {
+  bytes element = 3;
 
-    // (Required) The encoded element to pass to the transform.
-    bytes element = 3;
+  // Lower bound on timestamps of elements that this PTransform
+  // will produce into each of its output PCollections when invoked on this
+  // element and input. The map is keyed by the local output name of the
+  // PTransform.
+  map<string, google.protobuf.Timestamp> output_watermarks = 4;
 
-    // Approximate lower bounds on timestamps of elements that this PTransform
-    // will produce into each of its output PCollections, when invoked on this
-    // element. Keyed by the transform's local output name.
-    map<string, int64> output_watermarks = 4;
+  // Represents an estimate for the amount of currently outstanding work.
+  message Backlog {
+    // This informs Runners on how to aggregate the backlog
+    // being reported across multiple active bundles. Backlogs
+    // are aggregated using the set of partitions.
+    //
+    // For example SplittableDoFn's which consume elements from:
+    //  * a globally shared resource such as a Pubsub queue should set this
+    //    to “”.
+    //  * a shared partitioned resource should use the partition identifier.
+    //  * a uniquely partitioned resource such as a file range should set this to
+    //    file name + start offset.
+    bytes partition = 1;
+
+    // The estimate for the backlog.
+    oneof value {
+      // Represents an estimate for the amount of outstanding work. Values
+      // compare lexicographically.
+      bytes bytes = 1000;
 
-    // Approximate fraction of all work of the current bundle (before split)
-    // represented by invoking this Application and its downstream applications.
-    // The sum of fraction_of_work between all primary_roots and residual_roots
-    // must add up to approximately 1.0.
-    google.protobuf.DoubleValue fraction_of_work = 5;
+      // Whether the backlog is unknown.
+      bool is_unknown = 1001;
+    }
   }
 
-  // An an Application should be scheduled after a delay.
-  message DelayedApplication {
-    // The delay in seconds (lower bound).
-    double delay_sec = 1;
+  // (Required) An estimate for the amount outstanding work related to
+  // this application.
+  Backlog backlog = 5;
 
-    // (Required) The application that should be scheduled.
-    Application application = 2;
-  }
+  // (Required) Whether this application potentially produces an unbounded
+  // amount of data. Note that this should only be set to BOUNDED if and
+  // only if the application is known to produce a finite amount of output.
+  //
+  // Note that this is different from the backlog as the backlog represents
+  // how much work there is currently outstanding.
+  org.apache.beam.model.pipeline.v1.IsBounded.Enum is_bounded = 6;
 
-  // Root applications that should replace the current bundle.
-  repeated Application primary_roots = 1;
+  // Contains additional monitoring information related to this application.
```

Comment:
"related to" is ambiguous; can this be more precise? Is this monitoring info produced by the application so far? Or baseline monitoring data to be augmented by the current application?

Also, I see the `ProcessBundleProgressResponse` also contains `monitoring_infos`; are one of these fields redundant?

Reply:
Added a link to a doc that describes these additional signals.

Tthese monitoring infos give insights into a specific application that while the ProcessBundleProgressResponse contains monitoring info that is the aggregate of many applications.